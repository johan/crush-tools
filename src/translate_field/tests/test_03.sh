test_number=03
description="external mappings"

infile="$test_dir/test.in"
mapping="$test_dir/test_$test_number.map"
expected="$test_dir/test_$test_number.expected"

cat > $mapping << 'END'
$mapping{a} = 'AYE';
$mapping{e} = 'EEE';
$mapping{code} = 'HACK';
END

subtest=1
outfile="$test_dir/test_$test_number.$subtest.actual"
$bin -f 1 -e $mapping "$test_dir/test.in" > "$outfile"
if [ $? -ne 0 ] ||
   [ "`diff -q $outfile $expected`" ]; then
  test_status $test_number $subtest "$description (index)" FAIL
  has_error=1
else
  test_status $test_number $subtest "$description (index)" PASS
  rm "$outfile"
fi

subtest=2
outfile="$test_dir/test_$test_number.$subtest.actual"
$bin -F Field-0 -e $mapping "$test_dir/test.in" > "$outfile"
if [ $? -ne 0 ] ||
   [ "`diff -q $outfile $expected`" ]; then
  test_status $test_number $subtest "$description (label)" FAIL
  has_error=1
else
  test_status $test_number $subtest "$description (label)" PASS
  rm "$outfile"
fi

if [ ! $has_error ]; then
  rm "$mapping"
fi
