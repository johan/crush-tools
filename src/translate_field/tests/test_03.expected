Field-0,Field-1,Field-2
AYE,b,c
EEE,f,g
h,,j
HACK,in,c
HACK,in,pl
HACK,in,sh
